tests/python/unittest/test_conv_int8_intel.py

```diff
@@ -0,0 +1,140 @@
+             (28, 28, 512, 256, 1, 1, 0, 0, 2, 2),
+             (14, 14, 256, 1024, 1, 1, 0, 0, 1, 1),
+             (28, 28, 512, 1024, 1, 1, 0, 0, 2, 2),
+             (14, 14, 1024, 256, 1, 1, 0, 0, 1, 1),
+             (14, 14, 1024, 512, 1, 1, 0, 0, 2, 2),
+             (7, 7, 512, 2048, 1, 1, 0, 0, 1, 1),
+             (14, 14, 1024, 2048, 1, 1, 0, 0, 2, 2),
+             (7, 7, 2048, 512, 1, 1, 0, 0, 1, 1)
+            ]
+
+
+target_name = 'llvm -mcpu=skylake-avx512'
+avx2_len = 16
+ctx = tvm.context(target_name, 0);
+
+def getShape(im_height, im_width, in_filter, out_filter, kh, kw, hpad, wpad,
+             hstride, wstride, outDtype):
+    ## Find shapes
+    dataShape = (1, in_filter/avx2_len, im_height, im_width, avx2_len)
+
+    if outDtype == 'int32':
+        if kh != 1:
+            kernelShape = (out_filter/avx2_len, in_filter/avx2_len, kh, kw, avx2_len/4, avx2_len, 4)
+        else:
+            kernelShape = (out_filter/avx2_len, in_filter/avx2_len, avx2_len/4, avx2_len, 4, kh, kw)
+    elif outDtype == 'float32':
+        if kh != 1:
+            kernelShape = (out_filter/avx2_len, in_filter/avx2_len, kh, kw, avx2_len, avx2_len)
+        else:
+            kernelShape = (out_filter/avx2_len, in_filter/avx2_len, avx2_len, avx2_len, kh, kw)
+    out_height = (im_height + 2 * hpad - kh) // hstride + 1
+    out_width = (im_width + 2 * wpad - kw) // wstride + 1
+    oShape = (1, out_filter/avx2_len, out_height, out_width, avx2_len)
+    return (dataShape, kernelShape, oShape)
+
+
+
+def run_inference(dataDtype, kernelDtype, outDtype, im_height, im_width, in_filter,
+             out_filter, kh, kw, hpad, wpad, hstride, wstride):
+
+    (dataShape, kernelShape, oShape) = getShape(im_height, im_width, in_filter,
+                                                out_filter, kh, kw, hpad, wpad,
+                                                hstride, wstride, outDtype)
+
+    # Create TVM placeholders
+    data = tvm.placeholder(dataShape, name='data', dtype=dataDtype);
+    kernel = tvm.placeholder(kernelShape, name='kernel', dtype=kernelDtype);
+
+    # Create the numpy arrays to be used for executing conv models
+    if dataDtype == 'float32':
+        a = tvm.nd.array(np.random.rand(*dataShape).astype(dtype=dataDtype), ctx);
+        b = tvm.nd.array(np.random.rand(*kernelShape).astype(dtype=kernelDtype), ctx);
+    else:
+        a = tvm.nd.array(np.random.randint(100, size=dataShape).astype(dataDtype));
+        b = tvm.nd.array(np.random.randint(100, size=kernelShape).astype(kernelDtype));
+        #a = tvm.nd.array(np.ones(dataShape, dtype='uint8'), ctx);
+        #b = tvm.nd.array(np.zeros(kernelShape, dtype='int8'), ctx);
+
+    # cOrig will be used for declaration ouptut
+    # cSch will be used for scheduled computation output
+    cOrig = tvm.nd.array(np.zeros(oShape, dtype=outDtype), ctx);
+    cSch = tvm.nd.array(np.zeros(oShape, dtype=outDtype), ctx);
+
+
+    with tvm.target.create(target_name):
```

Comment:
I understand that quantization currently only works for x86 conv2d, so you want to invoke this "specialized" function directly. In the long team, I think the annotation I am doing could help here if quantization works on more devices. You can annotate node with the target and call conv2d_nchwc from a layer higher so that the dispatcher could find the correct compute.

Reply:
Agreed. This is a specific usecase to trigger x86 conv2d compute/schedule.